src/components/Workspace/Workspace.tsx

```diff
@@ -99,25 +126,35 @@ export default function Workspace(): JSX.Element {
     }
   }, [isDesktop, mobileActiveTab, layoutEditors]);
 
-  const [problemData, setProblemData] = useState<
-    ProblemData | null | undefined
-  >(undefined);
   const [problemID, setProblemID] = useState<string | null>(null);
   const [statusData, setStatusData] = useState<StatusData | null>(null);
 
   const updateProblemData = async (url: string | undefined) => {
     const newProblemID = usacoProblemIDfromURL(url);
+    if (newProblemID === problemID) return;
     setProblemID(newProblemID);
-    setProblemData(undefined);
     setStatusData(null);
+    const newJudgeResults = judgeResults;
+    while (newJudgeResults.length > 1) newJudgeResults.pop();
+
     if (newProblemID) {
       const newProblemData = await fetchProblemData(newProblemID);
+      const samples = newProblemData?.samples;
       setProblemData(newProblemData);
+      setJudgeResults(
+        resizeResults(newJudgeResults, 2 + (samples?.length ?? 0))
+      );
+      setInputTab('judge');
+    } else {
+      setProblemData(undefined);
+      setJudgeResults(newJudgeResults);
     }
   };
   useEffect(() => {
     updateProblemData(settings.judgeUrl);
-  }, [settings?.judgeUrl]);
+  });
```

Comment:
why remove `[settings?.judgeUrl]`? Won't doing this make the function run on every render (which seems bad for performance)?